### PR TITLE
Endpoints generator WIP

### PR DESCRIPTION
**Pull request by SimonLab**
This PR allow us to open the discussion on how we should implement the abase plugin, please read the last section (first version).

@nelsonic @jrans @eliascodes @shouston3 comments and issues are more than welcome!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,17 +17,25 @@
...
     "tape": "^4.6.0"
   },
   "scripts": {
-    "test": "node_modules/.bin/tape ./test/*.test.js",
+    "example": "node ./example/start.js",
+    "test": "node_modules/.bin/tape ./test/**/*.test.js",
     "lint": "node_modules/.bin/eslint .",
     "lint:fix": "node_modules/.bin/eslint . --fix",
-    "coverage": "node_modules/.bin/istanbul cover npm run test"
+    "coverage": "node_modules/.bin/istanbul cover ./node_modules/tape/bin/tape ./test/*.test.js && istanbul check-coverage --statements 100 --functions 100 --lines 100 --branches 100"
```
Comment (jrans) [inline package.json:30]: why remove the istanbul.yml and replace with something hard to read in the package.json without a target folder for coverage i.e. `src/`. Having a file for the config is much clearer and make adding options and understanding them more helpful than command line configuration? @nelsonic you think otherise?

File: lib/helpers/user/create_user.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+var path = require('path');
+
+module.exports = function(payload) {
+  var user = {};
+  user.id = users.length;
+  Object.keys(payload).forEach(function(key){
+    user[key] = payload[key];
+  })
+  users.push(user);
+  fs.writeFileSync(path.join(__dirname, '../../../users.json'), JSON.stringify(users));
```
Comment (jrans) [inline lib/helpers/user/create_user.js:12]: if this is occurring then `users.json` should be added to gitignore

File: lib/helpers/configuration/configuration.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+});
+
+config.push({
+  path: '/config/user',
+  method: 'post',
+  config: {
+    handler: function (request, reply) {
+      var newField = {};
+      existingFields[request.payload.field_name] = {type: request.payload.field_type};
+      fs.writeFileSync(path.join(__dirname, '../../fields.json'), JSON.stringify(existingFields), 'utf8');
```
Comment (jrans) [inline lib/helpers/configuration/configuration.js:30]: '../../fields.json' should be added to .gitignore for now?

File: example_config_2.json [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+          "field": "username",
+          "attributes": {
+            "name": "username",
+            "label": "Username: "
+          }
+        },
+        {
+          "field": "age",
+          "attributes": {
+            "name": "age",
```
Comment (jrans) [inline example_config_2.json:127]: @SimonLab as discussed `name` can be removed and field property should be the unique field identifier, which can infer its type from fields.json object

File: README.md [hunk 5]
```diff
@@ -79,3 +73,52 @@ a well-managed PostgreSQL cluster copes very well.
 > If you're still Curious or Worried about scaling PostgreSQL?
 see: https://www.citusdata.com
 > Want to model the network of people as a graph? https://github.com/cayleygraph/cayley
+
```
Comment (jrans) [inline README.md:76]: 👍

File: example/server.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+var hapi = require('hapi');
+var home = require('./home');
+var abase = require('../lib/index');
+
+exports.init = function (port, next) {
+
+  var server = new hapi.Server();
+  server.connection({port: port});
+  server.register([home, {register: abase, options: require('../example_config_2.json')}], function (err) {
```
Comment (jrans) [inline example/server.js:9]: requires should be at the top?

**Review by jrans — CHANGES_REQUESTED** [review]
small review for now

**Comment by samhstn** [comment]
Closed as too large of a pr, working from pr #41

**Pull request closed without merging**